### PR TITLE
add retrain for searched architetcure by ENAS

### PR DESCRIPTION
**Pull request by JiaojiaoYe1994**
I have implemented the retrain part PyTorch . It would be nice with feedback.

Env:
* nni 2.1
* torch 1.4
* linux
* local

To start the retrain process, here are examples:

* For **macro search**, assuming searched model is saved in `./macro.json`, so in './examples/nas/ens', run:
`python retrain.py --retrain_for macro --output_dir output_macro`
*  For **macro search**, assuming searched model is saved in `./micro.json`, so in './examples/nas/ens' run:
`python retrain.py --retrain_for micro --output_dir output_micro`

### CODE REVIEW

**Comment by ultmaster** [comment]
Please fix the lint.

File: examples/nas/enas/retrain.py [hunk 1]
```diff
@@ -0,0 +1,435 @@
+'''Script for ENAS retrain in nni using pytorch
```
Comment (ultmaster) [inline examples/nas/enas/retrain.py:1]: Please remove the lines in the header.

File: examples/nas/enas/retrain.py [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+
+from macro import GeneralNetwork
+from micro import MicroNetwork
+from nni.nas.pytorch.fixed import apply_fixed_architecture
+import datasets
+from utils import accuracy, reward_accuracy
+from nni.nas.pytorch.utils import AverageMeterGroup, to_device
+import json
+
+# TODO:
```
Comment (ultmaster) [inline examples/nas/enas/retrain.py:31]: What are the TODOs?

File: examples/nas/enas/retrain.py [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+aux_weight = 0.4
+result = {'accuracy':[],}
+def dump_global_result(res_path,global_result, sort_keys = False):
+    with open(res_path, "w") as ss_file:
+        json.dump(global_result, ss_file, sort_keys=sort_keys, indent=2)
+
+def main():
+    parse_args()
+
+    os.environ['CUDA_VISIBLE_DEVICES'] = '4'
```
Comment (ultmaster) [inline examples/nas/enas/retrain.py:414]: Please remove this. This should be done in bash script.

File: examples/nas/enas/retrain.py [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+    # micro search
+    elif FLAGS.retrain_for == 'micro':
+        print(' ** Micro search **')
+        child_model = MicroNetwork(num_layers=6, out_channels=20, num_nodes=5, dropout_rate=0.1, use_aux_heads=True)
+        apply_fixed_architecture(child_model, './micro.json')
+
+    train(child_model, device)
+
+if __name__ == "__main__":
+    main()
```
Comment (ultmaster) [inline examples/nas/enas/retrain.py:434]: Could you provide some reproduction results?

File: examples/nas/enas/retrain.py (hunk 3 above)
Comment (JiaojiaoYe1994) [reply examples/nas/enas/retrain.py:414]: Sure. Could you please tell me how to?

Comment (ultmaster) [reply examples/nas/enas/retrain.py:414]: For example, `CUDA_VISIBLE_DEVICES=4 python retrain.py`

**Comment by QuanluZhang** [comment]
closed due to no further update

**Pull request closed without merging**